src/Processors/Transforms/InflatingExpressionTransform.cpp

```diff
@@ -52,12 +52,23 @@ void InflatingExpressionTransform::transform(Chunk & chunk)
 Block InflatingExpressionTransform::readExecute(Chunk & chunk)
 {
     Block res;
-    if (likely(!not_processed))
+
+    if (!not_processed)
     {
-        res = getInputPort().getHeader().cloneWithColumns(chunk.detachColumns());
+        if (chunk.hasColumns())
+            res = getInputPort().getHeader().cloneWithColumns(chunk.detachColumns());
+
         if (res)
             expression->execute(res, not_processed, action_number);
     }
+    else if (not_processed->empty()) /// There's not processed data inside expression.
```

Comment:
@KochetovNicolai could you look at this function, please

Reply:
LGTM